dev/tasks/tasks.yml

```diff
@@ -941,6 +941,10 @@ tasks:
   test-r-devdocs:
     ci: github
     template: r/github.devdocs.yml
+    
+  test-r-depsource:
+    ci: github
+    template: r/github.linux.libsource.yml
```

Comment:
I'm not sure we need a custom .yml for this, I think you can get the desired result by using an existing docker image we test on and adding 

```
flags: '-e ARROW_DEPENDENCY_SOURCE=AUTO'
```

like `test-r-gcc-11` below

Reply:
Will do!